### PR TITLE
fix: various bugfixes

### PR DESCRIPTION
**Pull request by apalchys**

### CODE REVIEW

File: server/src/services/score.service.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    .leftJoin('mentor.user', 'mu')
+    .addSelect(getPrimaryUserFields('mu'))
+    .where('student."courseId" = :courseId', { courseId });
+
+  if (filter.activeOnly) {
+    query = query.andWhere('student."isFailed" = false').andWhere('student."isExpelled" = false');
+  }
+
+  if (filter.name) {
+    query = query.andWhere('("user"."firstName" ILIKE :searchText OR "user"."lastName" ILIKE :searchText)', {
```
Comment (apalchys) [inline server/src/services/score.service.ts:59]: fix is to add brackets @DmitryRomaniuk

Comment (DimaRomaniuk) [reply server/src/services/score.service.ts:59]: thx

File: client/src/pages/profile/index.tsx [hunk 2]
```diff
@@ -311,8 +311,10 @@ export class ProfilePage extends React.Component<Props, State> {
   };
 
   async componentDidMount() {
-    await this.fetchData();
-    await this.authorizeDiscord();
+    setTimeout(async () => {
```
Comment (DimaRomaniuk) [inline client/src/pages/profile/index.tsx:314]: Why do you use here setTimeout ?

Comment (apalchys) [reply client/src/pages/profile/index.tsx:314]: Thanks. I'll add a comment. It's a hack to fix some next.js caveat that router.query params are not available immediately on the client.

**Pull request merged**